tests/test_internal_functions.py

```diff
@@ -1,40 +1,95 @@
+        """),
+        "expected_tasks": [
+            ("system_schema", "tables", 1),
+            ("system_schema", "columns", 3),
+            ("system_schema", None, 4),
+            ("keyspace1", "standard1", 3),
+            ("keyspace1", None, 3),
+            (None, None, 7),
+            ("keyspace1x", None, 0),
+            ("keyspace1x", "table1x", 0),
+        ]
+    },
+    {
+        "id": "only_in_progress",
+        "output": textwrap.dedent("""\
+            pending tasks: 0
 
-    for output in [
-        '''
-        pending tasks: 6
-        - system_schema.tables: 1
-        - system_schema.columns: 2
-        - keyspace1.standard1: 3
-        ''',
-        '''
-        pending tasks: 6
-        - system_schema.tables: 1
-        - system_schema.columns: 2
-        - keyspace1.standard1: 3
+            id                                   compaction type keyspace      table   completed total unit progress
+            8e1f2d90-a252-11ee-a7f4-1bf9ae4e6ffd COMPACTION      system_schema columns 1         640   keys 0.16%
+            Active compaction remaining time :        n/a\
+        """),
+        "expected_tasks": [
+            ("system_schema", "tables", 0),
+            ("system_schema", "columns", 1),
+            ("system_schema", None, 1),
+            ("keyspace1", "standard1", 0),
+            ("keyspace1", None, 0),
+            (None, None, 1),
+            ("keyspace1x", None, 0),
+            ("keyspace1x", "table1x", 0),
+        ]
+    },
+    {
+        "id": "no_tasks",
+        "output": textwrap.dedent("""\
+            pending tasks: 0
+            \
+        """),
+        "expected_tasks": [
+            ("system_schema", "tables", 0),
+            ("system_schema", "columns", 0),
+            ("system_schema", None, 0),
+            ("keyspace1", "standard1", 0),
+            ("keyspace1", None, 0),
+            (None, None, 0),
+            ("keyspace1x", None, 0),
+            ("keyspace1x", "table1x", 0),
+        ]
+    }
+]
 
-        id                                   compaction type keyspace      table   completed total unit progress
-        8e1f2d90-a252-11ee-a7f4-1bf9ae4e6ffd COMPACTION      system_schema columns 1         640   
-    keys 0.16%
-        Active compaction remaining time :        n/a
-        '''
-    ]:
-        verify_cases(output, [
-                        ('system_schema', 'tables', 1),
-                        ('system_schema', 'columns', 2),
-                        ('system_schema', None, 3),
-                        ('keyspace1', 'standard1', 3),
-                        ('keyspace1', None, 3),
-                        (None, None, 6),
-                        ('keyspace1x', None, 0),
-                        ('keyspace1x', 'table1x', 0),
-                     ])
+@pytest.mark.parametrize("test_case", test_cases, ids=[tc["id"] for tc in test_cases])
+def test_parse_tasks(test_case):
+    output = test_case["output"]
```

Comment:
fixed https://github.com/scylladb/scylla-ccm/pull/611/commits/d25fb7a318d1c807ae216ed64174b3dd28aa2e68

Reply:
Why not change the `test_cases` directly to have `pytest.param`?

Reply:
Then `test_cases` would look like 

```
test_cases = [
    pytest.param(
        textwrap.dedent("""\
            pending tasks: 6
            - system_schema.tables: 1
            - system_schema.columns: 2
            - keyspace1.standard1: 3
            \
        """),
        [
            ("system_schema", "tables", 1),
            ("system_schema", "columns", 2),
            ("system_schema", None, 3),
            ("keyspace1", "standard1", 3),
            ("keyspace1", None, 3),
            (None, None, 6),
            ("keyspace1x", None, 0),
            ("keyspace1x", "table1x", 0),
        ],
        id="only_pending"
    ),
...
]
```
Which I don't know if it's better.
I like having the more explicit naming of everything, because `pytest.param` relies of order.

Reply:
That actually looks much better to me, it is exactly the structure the pytest.parametrize should have in my opinion and thus making it easier to understand in my opinion

Reply:
Both are o.k. from my POV

It's complex regardless of how you might present it.

Just a reference for issue in pytest about trying to introduce kwargs support for pytest.param, which never happened (but external plugging that supply something like it)
https://github.com/pytest-dev/pytest/issues/9216#issuecomment-1397286332